source/common/conn_pool/conn_pool_base.cc

```diff
@@ -534,17 +585,19 @@ void ConnPoolImplBase::onConnectionEvent(ActiveClient& client, absl::string_view
     // At this point, for the mixed ALPN pool, the client may be deleted. Do not
     // refer to client after this point.
     onConnected(client);
-    if (streams_available) {
+    if (client.readyForStream()) {
       onUpstreamReady();
     }
     checkForIdleAndCloseIdleConnsIfDraining();
     break;
   }
   case Network::ConnectionEvent::ConnectedZeroRtt: {
+    ENVOY_CONN_LOG(debug, "0-RTT connected with capacity {}", client,
+                   client.currentUnusedCapacity());
     // No need to update connecting capacity and connect_timer_ as the client is still connecting.
     ASSERT(client.state() == ActiveClient::State::Connecting);
     host()->cluster().stats().upstream_cx_connect_with_0_rtt_.inc();
-    // TODO(danzh) mark the client able to handle early data requests.
+    transitionActiveClientState(client, ActiveClient::State::ReadyForEarlyData);
```

Comment:
do we need to check capacity here?  can it be busy to start with?

Reply:
I'm not talking about a cache, I'm asking if a QUIC connection can start out without available streams.  may be a question for @RyanTheOptimist  :-)

Reply:
If the initial_max_streams_bidi transport param received in the handshake is 0, then no (bidirectional) streams may be opened. HTTP/3 recommends not doing this, of course, but it's definitely possible to start out at 0.

```
In order to permit these streams to open, an HTTP/3 server SHOULD configure non-zero minimum values for the number of permitted streams and the initial stream flow control window. So as to not unnecessarily limit parallelism, at least 100 request streams SHOULD be permitted at a time.
```

Reply:
So if the previous connection caches a SETTING or transport param with initial_max_streams_bidi = 0, the early data connection can start as BUSY. Is my understanding right?

Reply:
Yes, I think that's right.